### PR TITLE
Add Save class

### LINKED ISSUE #23: As a user, I can retrieve previously saved information from a data file
... so that I know the list of items I already have in my fridge.

### LINKED ISSUE #24: As a user, I can save information to a data file
... so that I can safely exit the system without losing information.

### PR DESCRIPTION
**Pull request by joohwan58**
I have added the save class, and also slightly edited the main class to facillitate saving.

I have also modified the toString method of the food class to suit my needs.

There is also an example of what the save file would look like.

### CODE REVIEW

**Comment by joohwan58** [comment]
Ah the checks are failing because the IO tests have not been updated

**Comment by joohwan58** [comment]
Right how can you account for the saving function in the ui test

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Save {
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java:19]: Class name should be a noun not a verb, can we name this Storage instead?

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public static String filePath = "save/savefile.txt";
+
+    /**
+     * Creates a textfile if it does not already exist.
+     */
+    public static void checkSave() {
+        Path path = Paths.get(filePath); //creates Path instance
+        try {
+            Path p = Files.createFile(path);     //creates file at specified location
+
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java]: Remove extra empty line

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        }
+    }
+
+    /**
+     * Reads the data from the textfile.
+     * @return a Fridge object constructed from the data in the savefile.
+     */
+    public static Fridge loadSave() {
+        Fridge fridge = new Fridge();
+        File f = new File(filePath);
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java]: Variable name should be 'file' not 'f'

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        try {
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String[] parameters = line.split(":");
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java:49]: Perhaps we can refactor this part into a separate function?

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                String name = parameters[1].substring(1, parameters[1].indexOf((",")));
+
+                String categoryStr = parameters[2].substring(1, parameters[2].indexOf((",")));
+                FoodCategory category = FoodCategory.convertStringToFoodCategory(categoryStr);
+
+                String expiry = parameters[3].substring(1, parameters[3].indexOf((",")));
+
+                String storageStr = parameters[4];
+                FoodStorageLocation storage = FoodStorageLocation.convertStringToLocation(storageStr);
+                fridge.add(new Food(category, name, expiry, storage));
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java]: I think this should be split into 2 lines

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            }
+        } catch (FileNotFoundException | InvalidDateException e) {
+            System.out.println("error loading save");
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java]: Would it be better to use the Ui class to print the error message.
Additionally, this magic string should be a variable message.
Perhaps we can also change the message to "There was an error loading the data for FridgeFriend! ):"

File: src/main/java/seedu/fridgefriend/FridgeFriend.java [hunk 7]
```diff
@@ -4,20 +4,24 @@
 import seedu.fridgefriend.exception.InvalidInputException;
 import seedu.fridgefriend.food.Fridge;
 import seedu.fridgefriend.utilities.Parser;
+import seedu.fridgefriend.utilities.Save;
 import seedu.fridgefriend.utilities.Ui;
 
 public class FridgeFriend {
 
     private static boolean isExit = false;
+    private static Fridge fridge = new Fridge();
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/FridgeFriend.java]: All the methods in Fridge are static, there is no need to declare a fridge variable here

File: src/main/java/seedu/fridgefriend/utilities/Save.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        } catch (IOException e) {
+            loadSave();
+        }
+    }
+
+    /**
+     * Reads the data from the textfile.
+     * @return a Fridge object constructed from the data in the savefile.
+     */
+    public static Fridge loadSave() {
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Save.java]: Method can return void, and access the Fridge statically.

**Review by kwokyto — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/fridgefriend/utilities/Save.java (hunk 1 above)
Comment (joohwan58) [reply src/main/java/seedu/fridgefriend/utilities/Save.java:19]: That conflicts with Food Storage Location, and save can be a noun anyway

**Pull request merged**